### PR TITLE
Removes all wrapped errors in materialize.go

### PR DESCRIPTION
**Pull request by epps**
# Description

Removes all wrapped errors in materialize.go

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: runner/materialize.go [hunk 1]
```diff
@@ -122,32 +122,29 @@ func (m MaterializeRunner) Run() (types.CompletionWatcher, error) {
 		m.Logger.Infow("Creating Index", "name", m.ID.Name, "variant", m.ID.Variant)
 		vectorStore, ok := m.Online.(provider.VectorStore)
 		if !ok {
-			return nil, fmt.Errorf("cannot create index on non-vector store: %v", m.Online)
-		}
-		if !ok {
-			return nil, fmt.Errorf("cannot create index on non-vector type: %v", m.VType)
+			return nil, fferr.NewInternalError(fmt.Errorf("cannot create index on non-vector store: %v", m.Online))
```
Comment (sdreyer) [inline runner/materialize.go:125]: is it possible to do m.Online.Type?

File: runner/materialize.go [hunk 2]
```diff
@@ -122,32 +122,29 @@ func (m MaterializeRunner) Run() (types.CompletionWatcher, error) {
...
 	m.Logger.Infow("Creating Table", "name", m.ID.Name, "variant", m.ID.Variant)
 	_, err = m.Online.CreateTable(m.ID.Name, m.ID.Variant, m.VType)
 	_, exists := err.(*fferr.DatasetAlreadyExistsError)
 	if err != nil && !exists {
-		return nil, fmt.Errorf("create table error: %w", err)
+		return nil, err
 	}
 	if exists && !m.IsUpdate {
-		return nil, fmt.Errorf("table already exists despite being new job")
+		return nil, fferr.NewDatasetAlreadyExistsError(m.ID.Name, m.ID.Variant, fmt.Errorf("table already exists despite being new job"))
```
Comment (sdreyer) [inline runner/materialize.go:139]: Should change to something like: "table x already exists" and print the table name

File: runner/materialize.go [hunk 3]
```diff
@@ -256,42 +253,42 @@ type MaterializedRunnerConfig struct {
 func (m *MaterializedRunnerConfig) Serialize() (Config, error) {
 	config, err := json.Marshal(m)
 	if err != nil {
-		panic(err)
```
Comment (sdreyer) [inline runner/materialize.go:259]: pain

File: runner/materialize.go (hunk 2 above)
Comment (epps) [reply runner/materialize.go:139]: There's not a clean way of getting the table name given the `OnlineStoreTable` interface only has `Get` and `Set` methods defined, but I'll change the error message.

Comment (sdreyer) [reply runner/materialize.go]: ah thats annoying

**Review by sdreyer — APPROVED** [review]

**Pull request merged**